WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java

```diff
@@ -280,35 +277,17 @@ private void changeLanguage(String languageCode) {
             return;
         }
 
-        Resources res = getResources();
-        Configuration conf = res.getConfiguration();
-        // will return conf.locale if conf is non-null, or Locale.getDefault()
-        Locale currentLocale = LanguageUtils.getCurrentDeviceLanguage(WordPress.getContext());
-        Locale newLocale = WPPrefUtils.languageLocale(languageCode);
-
-        if (currentLocale.toString().equals(newLocale.getDisplayLanguage())) {
+        if (!LocaleManager.isDifferentLanguage(languageCode)) {
             return;
         }
 
-        if (Locale.getDefault().toString().equals(newLocale.toString())) {
-            // remove custom locale key when original device locale is selected
-            mSettings.edit().remove(LANGUAGE_PREF_KEY).apply();
```

Comment:
I don't think we need this `mSettings` property anymore after these changes. Should we remove that as well?

Reply:
👍 Removed